File: dataset/human/java/file_272.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		long min = (long) Math.pow(10, 9);
		boolean minus = false;
		boolean zero = false;
		long ans = 0;
		for(int i = 0; i < N; i++) {
			long tmp = sc.nextLong();
			ans += Math.abs(tmp);
			min = Math.min(min, Math.abs(tmp));
			if(tmp < 0) {
				minus = !minus;
			} else if(tmp == 0) {
				zero = true;
			}
		}
		if((!zero) && minus) {
			ans -= min * 2;
		}
		System.out.println(ans);
	}

}

File: dataset/human/cpp/file_272.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main(){
  int ans;
  ans = 0;
  int r, g, b, N;
  cin >> r >> g >>  b >> N;
  
  for(int i = 0; i * r <= N; i++){
    for(int j = 0; j * g <= N - i*r; j++){
      if((N - i*r - j*g) % b == 0)ans++;
    }
  }
  
  cout << ans << endl;
}
